src/main/java/com/budgetapp/services/UserService.java

```diff
@@ -32,4 +32,9 @@ public User findUserById(final Long id) {
     public void deleteUser(final Long id) {
         userRepository.deleteById(id);
     }
+
+    public User authenticatingUser(final String username, final String password) {
+        final User user = userRepository.findByUsernameAndPassword(username, password).orElseThrow(() -> new EntityNotFoundException(String.format("")));
```

Comment:
Your message is completely pointless?

```suggestion
        final User user = userRepository.findByUsernameAndPassword(username, password).orElseThrow(EntityNotFoundException::new);
```

Reply:
.orElse(null) used this instead as EntityNotFoundException is a custom exception that returns 404